src/net.cpp

```diff
@@ -115,7 +117,10 @@ bool fDiscover = true;
 bool fListen = true;
 GlobalMutex g_maplocalhost_mutex;
 std::map<CNetAddr, LocalServiceInfo> mapLocalHost GUARDED_BY(g_maplocalhost_mutex);
-static bool vfLimited[NET_MAX] GUARDED_BY(g_maplocalhost_mutex) = {};
+
+static Synced<std::unordered_set<Network>> g_reachable_networks{
+    NET_UNROUTABLE, NET_IPV4, NET_IPV6, NET_ONION, NET_I2P, NET_CJDNS, NET_INTERNAL};
```

Comment:
My reasoning is that the most natural way to express this is to have a list of networks that are reachable (or unreachable if the logic is to be inverted) and `std::set` is the best way to do that. Having an array of `bool`s is a bit hackish as it requires that all enums are sequential and start at `0` and requires the existence of `NET_MAX` which smells bad. My plan is to remove `NET_MAX`. Ideally, enum lables should be used without depending on their specific integer values.

Anyway, if this is controveral I will ditch that change and leave it as `vfLimited[NET_MAX]` since it is not the main topic of this PR. Should I?

Reply:
enums are sequential and automatically start at 0; and `NET_MAX` is added for precisely that purpose. That's the way these things are designed to be used, it's not hackish or a bad smell...

Reply:
I disagree, but that discussion is for another PR. I reverted it back to what it is in `master` - an array of `bool`s.